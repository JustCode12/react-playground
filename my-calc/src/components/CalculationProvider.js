import React, { useState } from 'react';

export const CalculationContext = React.createContext();

const CalculationProvider = props => {
  const [result, setResult] = useState(0);

  const calculate = expression =>{
      setResult(eval(expression))
  }

  return (
    <CalculationContext.Provider
      value={{
        calculate,
        result
      }}
    >
      {props.children}
    </CalculationContext.Provider>
  );
};

export default CalculationProvider;