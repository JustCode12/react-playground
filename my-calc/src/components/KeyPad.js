import React, {useState, useContext} from 'react';
import { CalculationContext } from './CalculationProvider';

const  KeyPad = () => {

    const [expression, setExpression] = useState("")
    const {calculate} = useContext(CalculationContext);

    const onClick = e => {
        const clicked = e.target.name;
        if(expression.length>0&&clicked==="="){
            calculate(expression)
            setExpression("")
        }
        else{
        setExpression(expression + clicked); 
        }
    }
    return (
    <div><span>Data: </span>{expression}
      <div>
                <button name="(" onClick={onClick}>(</button>
                <button name="CE" onClick={onClick}>CE</button>
                <button name=")" onClick={onClick}>)</button>
                <button name="C" onClick={onClick}>C</button><br/>


                <button name="1" onClick={onClick}>1</button>
                <button name="2" onClick={onClick}>2</button>
                <button name="3" onClick={onClick}>3</button>
                <button name="+" onClick={onClick}>+</button><br/>


                <button name="4" onClick={onClick}>4</button>
                <button name="5" onClick={onClick}>5</button>
                <button name="6" onClick={onClick}>6</button>
                <button name="-" onClick={onClick}>-</button><br/>

                <button name="7" onClick={onClick}>7</button>
                <button name="8" onClick={onClick}>8</button>
                <button name="9" onClick={onClick}>9</button>
                <button name="*" onClick={onClick}>x</button><br/>


                <button name="." onClick={onClick}>.</button>
                <button name="0" onClick={onClick}>0</button>
                <button name="=" onClick={onClick}>=</button>
                <button name="/" onClick={onClick}>รท</button><br/>
            </div>
    </div>
    
    );

}

export default KeyPad;