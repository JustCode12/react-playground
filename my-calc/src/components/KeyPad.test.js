import React from 'react';
import { shallow, mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import KeyPad from './KeyPad.js';
import { CalculationContext } from './CalculationProvider';

configure({ adapter: new Adapter() })

const TestProvider = props => {
    const result = 5;
    return (
      <CalculationContext.Provider
        value={{ result }}>
        {props.children}
      </CalculationContext.Provider>
    )
  }
  describe('KeyPad component test', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(
        <TestProvider><KeyPad /></TestProvider>
      );
      expect(wrapper).toMatchSnapshot();
    });
    it('should be one KeyPad component', () => {
      const wrapper = mount(<TestProvider><KeyPad /></TestProvider>);
      expect(wrapper.find(KeyPad).length).toBe(1);
    })
    it('should have 20 buttons', () => {
        const wrapper = mount(<TestProvider><KeyPad /></TestProvider>);
        expect(wrapper.find('button').length).toBe(20);
      })
  });