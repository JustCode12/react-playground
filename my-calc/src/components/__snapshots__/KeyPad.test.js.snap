// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad component test should match the snapshot 1`] = `ShallowWrapper {}`;
